templates/devfiles/gradle/devfile.yaml

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2021 IBM Corporation and others
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+schemaVersion: 2.0.0
+metadata:
+  name: java-openliberty-gradle
+  version: 0.6.0
+  description: Java application Gradle built stack using the Open Liberty runtime
+  alpha.build-dockerfile: "{{.OUTERLOOP_DOCKERFILE_GRADLE_LOC}}"
+  alpha.deployment-manifest: "{{.DEVFILE_DEPLOY_YAML_GRADLE_LOC}}"
+starterProjects:
+  - name: user-app
+    git:
+      remotes:
+        origin: 'https://github.com/OpenLiberty/application-stack-starters.git'
+components:
+  - name: devruntime
+    container:
+      image: {{.STACK_IMAGE_GRADLE}}
+      memoryLimit: 1512Mi
+      mountSources: true
+      endpoints:
+        - exposure: public
+          path: /
+          name: ep1
+          targetPort: 9080
+          protocol: http
+commands:
+  - id: build
+    exec:  
+      component: devruntime 
+      commandLine: /stack/ol/scripts/gradle/devbuild-cmd.sh
```

Comment:
Don't we have to pass the Liberty version or no?

Reply:
No, not really, The assemble process does not require/use the liberty driver level. It just gets deps and builds the project. It does nothing with the liberty plugin/tasks.

Reply:
So you're saying gradle assemble is only ever going to build a WAR?  It's never going to do anything like deploy to Liberty ? In the analogous Maven case.. mvn package might install a server if you use something like this:  https://github.com/OpenLiberty/ci.maven/blob/master/liberty-maven-app-parent/pom.xml   so it's arguably worth including the version.

Reply:
ok yeah i see what you are saying. a gradle build file could have something like assemble.finalizedBy 'libertyCreate' which would download/install the latest driver. Will change that.